packages/yarn-plugin/src/util.ts

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { PortablePath, ppath, xfs } from '@yarnpkg/fslib';
+import { valid as semverValid } from 'semver';
+import { getManifestByVersion } from '@backstage/release-manifests';
+import { Descriptor, structUtils } from '@yarnpkg/core';
+import { PROTOCOL } from './constants';
+
+const isWorkspaceRoot = (dir: PortablePath) => {
+  const manifestPath = ppath.join(dir, 'package.json');
+
+  if (xfs.existsSync(manifestPath)) {
+    const manifest = xfs.readJsonSync(manifestPath);
+
+    if (manifest.workspaces) {
+      return true;
+    }
+  }
+
+  return false;
+};
+
+const findWorkspaceRoot = () => {
```

Comment:
Explore using `cli-common` here instead. It's a very lightweight package

Reply:
Done in 573e98b352484183a8c96e37aad7f771bcd9235c